src/api.py

```diff
@@ -42,59 +64,199 @@ def config_cls(cls) -> Type[Config]:
     config: PromptGenerationPluginConfig
 
     def __init__(
-        self,
-        client: Steamship = None,
-        config: Dict[str, Any] = None,
-        context: InvocationContext = None,
+            self,
+            client: Steamship = None,
+            config: Dict[str, Any] = None,
+            context: InvocationContext = None,
     ):
         super().__init__(client, config, context)
         openai.api_key = self.config.openai_api_key
 
+    def completion_with_retry(self, **kwargs: Any) -> Any:
+        """Use tenacity to retry the completion call."""
+        min_seconds = 4
+        max_seconds = 10
```

Comment:
is the implication here (with default params) that _at most_ we expect up to 1 min of retrying in the `run` ?